commit d02df4bcd1aee3ed287ee8f8c08c34c47db4cd1e
Author: Andrew Price <anprice@redhat.com>
Date:   Wed Jun 15 13:09:41 2011 +0100

    gfs2_edit: Fix savemeta compression for older zlibs
    
    gzdopen in zlib 1.2.3 parses the mode string in a different way to
    subsequent versions and the mode string we use causes the older gzwrite
    to fail with EBADF. This patch fixes the mode string so that the
    gzwrites succeed with the old and new zlibs.
    
    rhbz#702313
    
    Reviewed-by: Steven Whitehouse <swhiteho@redhat.com>
    Signed-off-by: Andrew Price <anprice@redhat.com>

diff --git a/gfs2/edit/savemeta.c b/gfs2/edit/savemeta.c
index cb7e243..160277c 100644
--- a/gfs2/edit/savemeta.c
+++ b/gfs2/edit/savemeta.c
@@ -209,7 +209,7 @@ static void warm_fuzzy_stuff(uint64_t wfsblock, int force)
 static struct metafd savemetaopen(char *out_fn, int gziplevel)
 {
 	struct metafd mfd;
-	char gzmode[5] = "rwb9";
+	char gzmode[3] = "w9";
 	char dft_fn[] = DFT_SAVE_FILE;
 
 	if (!out_fn) {
@@ -232,7 +232,7 @@ static struct metafd savemetaopen(char *out_fn, int gziplevel)
 
 	mfd.gziplevel = gziplevel;
 	if (gziplevel > 0) {
-		gzmode[3] = '0' + gziplevel;
+		gzmode[1] = '0' + gziplevel;
 		mfd.gzfd = gzdopen(mfd.fd, gzmode);
 		if (!mfd.gzfd) {
 			fprintf(stderr, "gzdopen error: %s\n", strerror(errno));
